docs/calendar-draft.md

```diff
@@ -0,0 +1,315 @@
+# Draft Design of Temporal Calendar API
+
+This doc describes a design for first-class support for non-Gregorian [calendars](https://en.wikipedia.org/wiki/Calendar) in Temporal.
+
+## Temporal.Date internal slots
+
+Temporal.Date currently has three internal slots: year, month, and day. (An "internal slot" refers to actual data, as opposed to "properties", which could be computed.)  In this proposal, three more internal slots would be added:
+
+1. calendar
+2. isLeapMonth
+3. era
+
+It is believed that a date in any known calendar can be described using the five-tuple (*year*, *month*, *day*, *isLeapMonth*, *era*).  Since all operations involving the slots of Temporal.Date will go through the calendar first, calendars that need more data can assign complex types to these slots.  It is possible that *isLeapMonth* can be merged into *month* by expanding the set of numbers that *month* can represent.
+
+The *calendar* slot contains an object implementing the Temporal.Calendar interface, described below.
+
+## Temporal.Calendar interface
+
+The new Temporal.Calendar interface is a mechanism to allow arbitrary calendar systems to be implemented on top of Temporal.  ***Most users will not encounter the Temporal.Calendar interface directly***, unless they are building or using a non-built-in calendar system.
+
+### Methods on the Temporal.Calendar interface
+
+All of the following methods return new Temporal.Date objects.  All properties of `Temporal.Calendar` are Symbols.
```

Comment:
Why would the properties be Symbols? I’d say they are most likely not.

Reply:
Also the statement “all calendar metjods return new Date objects” is certainly false.

They return Date/DateTime/Time objects (and potentially Absolutes as well)

Reply:
Fixed the language about the return types.  Putting the discussion about symbols in the other thread with ljharb.